app/ide-desktop/client/tests/electronTest.ts

```diff
@@ -0,0 +1,47 @@
+/** @file Commonly used functions for electron tests */
+
+import { _electron, expect, type Page, test } from '@playwright/test'
+
+/**
+ * Tests run on electron executable.
+ *
+ * Similar to playwright's test, but launches electron, and passes Page of the main window.
+ */
+export function electronTest(name: string, body: (page: Page) => Promise<void> | void) {
+  test(name, async () => {
+    const app = await _electron.launch({
+      executablePath: process.env.ENSO_TEST_EXEC_PATH ?? '',
+      env: { ...process.env, ['ENSO_TEST']: name },
+    })
+    const page = await app.firstWindow()
+    await body(page)
+    await app.close()
+  })
+}
+
+/**
+ * Login as test user. This function asserts that page is the login page, and uses
+ * credentials from ENSO_TEST_USER and ENSO_TEST_USER_PASSWORD env variables.
+ */
+export async function loginAsTestUser(page: Page) {
+  // Login screen
+  await expect(page.getByRole('textbox', { name: 'email' })).toBeVisible()
+  await expect(page.getByRole('textbox', { name: 'password' })).toBeVisible()
+  if (process.env.ENSO_TEST_USER == null || process.env.ENSO_TEST_USER_PASSWORD == null) {
+    throw Error(
+      'Cannot log in; ENSO_TEST_USER and ENSO_TEST_USER_PASSWORD env variables are not provided',
+    )
+  }
+  await page.keyboard.insertText(process.env.ENSO_TEST_USER)
+  await page.keyboard.press('Tab')
+  await page.keyboard.insertText(process.env.ENSO_TEST_USER_PASSWORD)
+  await page.keyboard.press('Enter')
+
+  // Accept terms screen
+  await expect(page.getByText('I agree')).toHaveCount(2)
```

Comment:
i suppose this is sufficient for now. but we will probably want to extract dashboard and/or GUI actions to a shared package